config/src/main/java/com/networknt/config/Config.java

```diff
@@ -244,9 +221,38 @@ public InputStream getInputStreamFromFile(String filename) {
             return getConfigStream(filename, "");
         }
 
+        /**
+         * Method used to load the configuration file as a given Object based on the config loader class configured in config.yml and cache it.
+         * If no config loader is configured, file will be loaded by default loading method.
+         * @param configName    The name of the config file, without an extension
+         * @param clazz         The class that the object will be deserialized into
+         * @param path          The relative directory or absolute directory that config will be loaded from
+         * @return An instance of the object if possible, null otherwise. IOExceptions smothered.
+         */
         @Override
         public Object getJsonObjectConfig(String configName, Class clazz, String path) {
-            checkCacheExpiration();
+            Object config = configCache.get(configName);
+            if (config == null) {
+                synchronized (FileConfigImpl.class) {
+                    config = configCache.get(configName);
+                    if (config == null) {
+                        config = loadJsonObjectConfigWithSpecificConfigLoader(configName, clazz, path);
```

Comment:
the configs for modules other than Config will use "getJsonObjectConfig", and load with different "ConfigLoader" defined in config.yml